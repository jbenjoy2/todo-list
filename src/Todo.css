.Todo-form {
	margin: 10px auto;
}

.Todo-task {
	margin: 10px 3px;
	visibility: visible;
	opacity: 1;
	transition: opacity .5s linear;
}

.Todo-completed {
	text-decoration: line-through;
}

.Todo-hidden {
	visibility: hidden;
	opacity: 0;
	transition: visibility 0s .5s, opacity .5s linear;
}

.Todo-buttons {
	margin: 10px 0px;
}

.Todo-btn {
	margin: auto 3px;
	border-radius: 5px;
	background-color: rgb(159, 29, 211);
	color: white;
	padding: 6px 3px;
	box-shadow: 0px 2px 1px 2px rgba(0, 0, 0, 0.24);
}
.Todo-btn:active {
	transform: translateY(2px);
	box-shadow: none;
}
