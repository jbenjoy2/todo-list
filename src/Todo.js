import React, { useState } from 'react';
import './Todo.css';

const Todo = ({ task = 'new task', id = '1', remove, update }) => {
	const [ updatedTask, setUpdatedTask ] = useState(task);
	const [ editing, setEditing ] = useState(false);

	const toggleEdit = () => {
		setEditing((edit) => !edit);
	};

	const handleChange = (e) => {
		setUpdatedTask(e.target.value);
	};

	const handleUpdate = (e) => {
		e.preventDefault();
		update(id, updatedTask);
		setEditing(false);
	};
	const handleRemove = () => remove(id);

	if (editing) {
		return (
			<div className="Todo-form">
				<form onSubmit={handleUpdate}>
					<input type="text" value={updatedTask} onChange={handleChange} autoFocus={true} />
					<button>Update</button>
				</form>
			</div>
		);
	}
	return (
		<div className="Todo-task">
			<li>{task}</li>
			<button onClick={toggleEdit}>Edit Task</button>
			<button onClick={handleRemove}>Delete Task</button>
		</div>
	);
};
export default Todo;
