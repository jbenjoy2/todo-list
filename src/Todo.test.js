import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Todo from './Todo';

// smoke test
it('renders without crashing', () => {
	render(<Todo />);
});

// snapshot test
it('matches snapshot', () => {
	const { asFragment } = render(<Todo />);
	expect(asFragment).toMatchSnapshot();
});

// editing snapshot test
it('matches editing snapshot', () => {
	const { asFragment, getByText } = render(<Todo />);
	const editBtn = getByText('Edit Task');
	fireEvent.click(editBtn);
	expect(asFragment).toMatchSnapshot();
});

it('runs task update function when edit form is submitted', () => {
	const mockEditFunc = jest.fn();

	const { getByText } = render(<Todo update={mockEditFunc} />);
	// get edit button to click it
	const editBtn = getByText('Edit Task');
	fireEvent.click(editBtn);
	const updateBtn = getByText('Update');
	fireEvent.click(updateBtn);
	expect(mockEditFunc).toHaveBeenCalled();
});

it('runs task delete function when delete button is clicked', () => {
	const mockDeleteFunc = jest.fn();
	const { getByText } = render(<Todo remove={mockDeleteFunc} />);
	const deleteBtn = getByText('Delete Task');
	fireEvent.click(deleteBtn);
	expect(mockDeleteFunc).toHaveBeenCalled();
});
