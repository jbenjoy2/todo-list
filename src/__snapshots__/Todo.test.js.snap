// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches editing snapshot 1`] = `[Function]`;

exports[`matches snapshot 1`] = `[Function]`;
